### ISSUE TITLE
Intersection Detection

### ISSUE THREAD

**Issue by Karljohan99** [TonyXuQAQ/RNGDetPlusPlus#25]
Hello, thanks for releasing the code. I trained the RNGDet++ model using data gathered from various European countries, and while the pixel accuracy was similar to the results reported in the paper, the intersection accuracy was significantly lower. I also tried fine-tuning the model on my small lane detection dataset, and the model couldn't detect almost any intersections. However, if I fine-tuned the backbone and the FPN head only with intersection data, the model could detect some intersections correctly. A possible cause for this might be that my data has higher pixel resolution, thus there are fewer intersections on each ROI sample, and the intersection loss is much smaller than the segment loss. I have thought about increasing the positive weight for the intersection loss, adding a third backbone for solely intersection detection and increasing the crop size of ROI. Would these ideas be worth trying? Do you have any other suggestions for improving the intersection detection accuracy?

Also, why are intersections detected at the edges of the satellite image, even if no actual intersections exist? Currently, these detections are filtered out in the code, which means that roads with no intersections on the satellite image won't be detected. These detections at the edges didn't appear when I fine-tuned the backbone separately with only intersection data.

Thank you!

**Comment by TonyXuQAQ** [comment]
Thanks for the question. We have not tested our model on images with higher resolution, which is an interesting problem to explore. If the resolution is higher,
1. For the segmentation of intersection points, you should tune the parameters to generate the GT segmentation label. Is the GT intersection segmentation label too sparse? You can increase the size of the intersection point.
2. For the agent, it is suggested to increase the ROI size and **agent step length**.

For your reference, you might want to find some papers about segmentation with sparse/unbalanced/small object segmentation labels.
